staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -220,14 +221,20 @@ func (c *AvailableConditionController) sync(key string) error {
 			availableCondition.Reason = "EndpointsNotFound"
 			availableCondition.Message = fmt.Sprintf("cannot find endpoints for service/%s in %q", apiService.Spec.Service.Name, apiService.Spec.Service.Namespace)
 			apiregistration.SetAPIServiceCondition(apiService, availableCondition)
-			_, err := updateAPIServiceStatus(c.apiServiceClient, originalAPIService, apiService)
+			if _, err2 := updateAPIServiceStatus(c.apiServiceClient, originalAPIService, apiService); err2 != nil {
+				klog.Errorf("cannot update API service status: %v", err2)
+				return err
```

Comment:
Did you mean err or err2? If the former, why include this at all? It will happen one line down.

Reply:
This changes the error that is returned.

Reply:
I removed the extra return of err.
In the log, err2 was intended - otherwise we may not know status update failed.

Reply:
@lavalamp 
Can you give this another look ?